magefiles/utils.go

```diff
@@ -287,3 +287,11 @@ func cleanupQuayTags(quayService quay.QuayService, organization, repository stri
 	}
 	return fmt.Errorf("encountered errors during CleanupQuayTags: %s", errBuilder.String())
 }
+
+// deletePreviewBranch deletes 'preview-' branch
+// used on ci job to delete 'preview-' branch that is created in the infra-deployment fork
+func deletePreviewBranch() {
+	if err := sh.RunV("bash", "-c", "git branch -D `git branch --list 'preview-*'`"); err != nil {
```

Comment:
hmm not sure if this will work. afaik this will only delete the branch on local git. for deleting the branch you'd need to [call github api](https://docs.github.com/en/rest/git/refs?apiVersion=2022-11-28#delete-a-reference)

would it make more sense to create a periodic github actions job in [redhat-appstudio-qe/infra-deployments](https://github.com/redhat-appstudio-qe/infra-deployments) repo that will delete preview-* branches older than ~1 month (stale branches are sometimes useful for debugging)?

Reply:
You are right :( Goingggg


I thought about it: https://issues.redhat.com/browse/RHTAPBUGS-570?focusedId=22955513&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-22955513. I was thinking of implementing both approaches to not accumulate the branches that are created by ci.

Reply:
cool. well the github action could take care of every stale branch, no? I'm not really sure of a benefit of adding a code for branch cleanup of infra-deployments repo to e2e-tests repo.

Reply:
The benefit is to not wait for ~1 month (or other date time range) to delete the preview branches of ci jobs that we are sure can be deleted

Reply:
hmm weren't we fine with thousands of branches living their life and not being deleted for more than a year? 🤪 

what I wanted to say is that adding a code for branch deletion here is just extra work that might not be needed, could cause issues in test runs and could require a maintenance in a future. let's wait for other people what they think

btw for the initial cleanup I think it's worth to consider nuking the whole [infra-deployments](https://github.com/redhat-appstudio-qe/infra-deployments) fork (during quiet period) rather than shelling the githup api with 20k+ branch delete requests 💥 
wdyt @flacatus @rhopp

Reply:
We discussed the cronjob for cleanup of the branches, but I was thinking more in lines of "each test should cleanup after themselves". It seems to me like a cleaner approach, even thought it adds some code to our codebase.
To some of your concerns:
* work that might not be needed - well... we either do it here, or we create the code somewhere (slightly) else (cron job)
* could cause issues in test runs - we need to ensure, that possibly failed cleanup doesn't fail the test (it could send an alert though)
* require a maintenance in a future - same thing can be said about the cron job solution as well.

I'm not *strictly* against cron job, but I have slight preference for tests cleaning up after themselves instead of bulk cleanup in cron job.
As for the initial cleanup I agree that nuking & recreating will be probably better.

Reply:
+1 To recreate infra-deployments and then have a cron job which deletes branches with 1 week life. Why? I was in case where i debug this branches locally :(

Reply:
+1 ^